### ISSUE TITLE
error event doesn't always emit an Error instance

### ISSUE THREAD

**Issue by niik** [aws/aws-iot-device-sdk-js#95]
See [index.js L744](https://github.com/aws/aws-iot-device-sdk-js/blob/97b0b468d59e02e2f6a1dce321000d914cb47894/device/index.js#L744)

```js
} else {
  that.emit('error', 'Maximum queued offline subscription operations reached.');
}
```

The [other occurance](https://github.com/aws/aws-iot-device-sdk-js/blob/97b0b468d59e02e2f6a1dce321000d914cb47894/device/index.js#L783) of the `error` event seems to be using `new Error(msg)` as does [mqtt](https://github.com/mqttjs/MQTT.js/search?utf8=✓&q=%27error%27+emit&type=Code) so I'm assuming this is a typo?

**Comment by fengsongAWS** [comment]
Hi @niik ,

Thanks for your feed back on javascript SDK. 
What you are proposing is probably correct.  We will look into that and have it in our backlog to fix that.   
In addition, do you run into any specific scenarios caused by this problem? If so, we would appreciate if you could elaborate on this. 

Thanks,
Fengyi

**Comment by niik** [comment]
> In addition, do you run into any specific scenarios caused by this problem? If so, we would appreciate if you could elaborate on this.

No problem per se, I've just started playing around with this library and I noticed that there were no typescript definitions for it so I wrote some (I've yet to submit them to DefinitelyTyped, I can update this issue when I've done so if you wish) and I spotted this inconsistency in the process.

Normally I would have submitted a PR but in lieu of a [contributing.md](https://github.com/blog/1184-contributing-guidelines) file I looked at previously submitted PRs and found [a comment](https://github.com/aws/aws-iot-device-sdk-js/pull/94#issuecomment-257957232) that I interpreted as you not accepting public contributions via pull requests and thus I opened this issue.

**Comment by niik** [comment]
> I've yet to submit them to DefinitelyTyped, I can update this issue when I've done so if you wish

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12556

**Comment by fengsongAWS** [comment]
Hi @niik ,

Thank you very much for your interest and efforts in AWS IoT Node.js SDK. Yes, we normally do not directly take PR because of the internal development process. However, we do give credit to people whose contributions were taken and incorporated into our internal pipeline test/release. 
We will keep watching this and have this issue in our backlog. We will fix it in our future release.

Thanks,
Fengyi

**Comment by liuszeng** [comment]
Issue addressed in release [v1.0.14](https://github.com/aws/aws-iot-device-sdk-js/releases/tag/v1.0.14).
